inference-engine/src/mkldnn_plugin/nodes/mkldnn_def_conv_node.cpp

```diff
@@ -267,218 +277,72 @@ struct jit_uni_def_conv_kernel_f32 : public jit_uni_def_conv_kernel, public jit_
-                        paddd(xmm_v4_off, xmm_w_high);
-
-
-                        movss(xmm_w1, xmm_hh);
-                        mulss(xmm_w1, xmm_hw);
+                        Xmm xmm_w1 = Xmm(4);
+                        Xmm xmm_w2 = Xmm(1);
+                        Xmm xmm_w3 = Xmm(8);
+                        Xmm xmm_w4 = Xmm(5);
+
+                        Xmm xmm_v1 = Xmm(2);
+                        Xmm xmm_v2 = Xmm(3);;
+                        Xmm xmm_v3 = Xmm(6);
+                        Xmm xmm_v4 = Xmm(7);
+
+                        Vmm vmm_w1 = Vmm(xmm_w1.getIdx());
+                        Vmm vmm_w2 = Vmm(xmm_w2.getIdx());
+                        Vmm vmm_w3 = Vmm(xmm_w3.getIdx());
+                        Vmm vmm_w4 = Vmm(xmm_w4.getIdx());
+
+                        Vmm vmm_v1 = Vmm(xmm_v1.getIdx());
+                        Vmm vmm_v2 = Vmm(xmm_v2.getIdx());
+                        Vmm vmm_v3 = Vmm(xmm_v3.getIdx());
+                        Vmm vmm_v4 = Vmm(xmm_v4.getIdx());
+
+                        // offsets computation
+                        size_t ind_off_hh = sampledPointsPerPixel * (((size_t) kh * jcp_.kw + kw) + ow * (jcp_.kh * jcp_.kw));
+                        size_t ind_off_hl = ind_off_hh + 1;
+                        size_t ind_off_lh = ind_off_hl + 1;
+                        size_t ind_off_ll = ind_off_lh + 1;
+
+                        movq(xmm_v1_off, qword[aux_reg_sampled_offs + ind_off_ll * jcp_.typesize_sampled_offsets]);
+                        movq(xmm_v2_off, qword[aux_reg_sampled_offs + ind_off_hl * jcp_.typesize_sampled_offsets]);
+                        movq(xmm_v3_off, qword[aux_reg_sampled_offs + ind_off_lh * jcp_.typesize_sampled_offsets]);
+                        movq(xmm_v4_off, qword[aux_reg_sampled_offs + ind_off_hh * jcp_.typesize_sampled_offsets]);
+
+                        // w's computation
+                        // w22 ~ h_high * w_high
+                        // w21 ~ h_high * w_low
+                        // w12 ~ h_low * w_high
+                        // w11 ~ h_low * w_low
+                        movd(xmm_w1, dword[aux_reg_sampled_wei + ind_off_ll * jcp_.typesize_sampled_wei]);
                         uni_vbroadcastss(vmm_w1, xmm_w1);
-
-                        movss(xmm_w2, xmm_hh);
-                        mulss(xmm_w2, xmm_lw);
+                        movd(xmm_w2, dword[aux_reg_sampled_wei + ind_off_hl * jcp_.typesize_sampled_wei]);
                         uni_vbroadcastss(vmm_w2, xmm_w2);
-
-                        movss(xmm_w3, xmm_lh);
-                        mulss(xmm_w3, xmm_hw);
+                        movd(xmm_w3, dword[aux_reg_sampled_wei + ind_off_lh * jcp_.typesize_sampled_wei]);
                         uni_vbroadcastss(vmm_w3, xmm_w3);
-
-                        movss(xmm_w4, xmm_lh);
-                        mulss(xmm_w4, xmm_lw);
+                        movd(xmm_w4, dword[aux_reg_sampled_wei + ind_off_hh * jcp_.typesize_sampled_wei]);
```

Comment:
In this context movd instruction looks unnecessary, since we can broadcast a value directly from memory `uni_vbroadcastss(vmm_w4, ptr[aux_reg_sampled_wei + ind_off_hh * jcp_.typesize_sampled_wei]);`